core/src/main/java/tc/oc/pgm/consumable/ConsumableMatchModule.java

```diff
@@ -0,0 +1,63 @@
+package tc.oc.pgm.consumable;
+
+import com.google.common.collect.ImmutableSet;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerItemConsumeEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.ParticipantState;
+import tc.oc.pgm.kits.tag.ItemTags;
+
+public class ConsumableMatchModule implements MatchModule, Listener {
+
+  private final Match match;
+  private final ImmutableSet<ConsumableDefinition> consumableDefinitions;
+
+  public ConsumableMatchModule(
+      Match match, ImmutableSet<ConsumableDefinition> consumableDefinitions) {
+    this.match = match;
+    this.consumableDefinitions = consumableDefinitions;
+  }
+
+  @EventHandler
+  private void onItemConsume(PlayerItemConsumeEvent event) {
+    ConsumableDefinition consumableDefinition = getConsumableDefinition(event.getItem());
+    if (consumableDefinition == null) return;
+    if (consumableDefinition.getCause() != ConsumeCause.EAT) return;
+
+    Player player = event.getPlayer();
+    MatchPlayer matchPlayer = match.getPlayer(player);
+    ParticipantState playerState = match.getParticipantState(player);
+    if (matchPlayer == null || playerState == null) return;
```

Comment:
```suggestion
    MatchPlayer matchPlayer = match.getPlayer(player);
    if (!MatchPlayers.canInteract(matchPlayer)) return;
```
there's no need for a participant state, and you can just check for interaction (the util in MatchPlayers handles the null-check for you)